alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/metrics/MetricsBean.java

```diff
@@ -107,6 +111,31 @@ public void doSuspended() {
     suspended.incrementAndGet();
   }
 
+  public void doRepositoryReceived() {
+    repositoryReceived.incrementAndGet();
+  }
+
+  public void doRepositoryAccepted() {
+    repositoryAccepted.incrementAndGet();
+  }
+
+  public void doRepositoryAccepted(int size) {
+    repositoryAccepted.getAndAdd(size);
+  }
+
+  public void doRepositoryRejected() {
+    repositoryReceived.decrementAndGet();
+    repositoryRejected.incrementAndGet();
+  }
+
+  public void doRepositoryAvgTime(long time) {
+    if (repositoryAvgTime.get() == 0) {
+      repositoryAvgTime.set(time);
+    } else {
+      repositoryAvgTime.set((repositoryAvgTime.get() + time) / 2);
```

Comment:
I think the algorithm here is wrong, you cannot get a new avg from  (avg1 + avg2) / 2.

You can only calculate a new avg from sum(old_time) + sum(new_time) / (old_count + new_count).

Reply:
> I think the algorithm here is wrong, you cannot get a new avg from (avg1 + avg2) / 2.
> 
> You can only calculate a new avg from sum(old_time) + sum(new_time) / (old_count + new_count).

I think we can use the moving average for the time being.